From 6eac88bef8cfc2cff39d0487cbf9d7743f4c10e4 Mon Sep 17 00:00:00 2001
From: Subhajit Paul <subhajit_paul@ti.com>
Date: Tue, 2 Jul 2019 16:31:19 +0530
Subject: [PATCH 1/7] arm64: dts: ti: add new DT overlays for automotive

k3-j721e-auto-common.dtso:
*   Memory map update for common automotive use cases

k3-j721e-vision-apps.dtso:
* Memory carveouts for vision use cases
* Move DSS and HDMI ownership to RTOS

k3-j721e-psdkla-apps.dtso:
* Disable uart from Linux to be used for gateway demos

Signed-off-by: Subhajit Paul <subhajit_paul@ti.com>
Signed-off-by: Nikhil Devshatwar <nikhil.nd@ti.com>
---
 arch/arm64/boot/dts/ti/Makefile                  |   5 +-
 arch/arm64/boot/dts/ti/k3-j721e-auto-common.dtso | 182 ++++++++++++++++++++++
 arch/arm64/boot/dts/ti/k3-j721e-psdkla-apps.dtso |  14 ++
 arch/arm64/boot/dts/ti/k3-j721e-vision-apps.dtso | 190 +++++++++++++++++++++++
 4 files changed, 390 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm64/boot/dts/ti/k3-j721e-auto-common.dtso
 create mode 100644 arch/arm64/boot/dts/ti/k3-j721e-psdkla-apps.dtso
 create mode 100644 arch/arm64/boot/dts/ti/k3-j721e-vision-apps.dtso

diff --git a/arch/arm64/boot/dts/ti/Makefile b/arch/arm64/boot/dts/ti/Makefile
index ea9474c..ea3fa30 100644
--- a/arch/arm64/boot/dts/ti/Makefile
+++ b/arch/arm64/boot/dts/ti/Makefile
@@ -26,7 +26,10 @@ dtb-$(CONFIG_ARCH_K3_J721E_SOC) += k3-j721e-common-proc-board.dtb \
 				   k3-j721e-common-proc-board-infotainment.dtbo \
 				   k3-j721e-common-proc-board-infotainment-display-sharing.dtbo \
 				   k3-j721e-common-proc-board-jailhouse.dtbo \
-				   k3-j721e-pcie-backplane.dtbo
+				   k3-j721e-pcie-backplane.dtbo \
+				   k3-j721e-auto-common.dtbo \
+				   k3-j721e-psdkla-apps.dtbo \
+				   k3-j721e-vision-apps.dtbo
 
 $(obj)/%.dtbo: $(src)/%.dtso FORCE
 	$(call if_changed_dep,dtc)
diff --git a/arch/arm64/boot/dts/ti/k3-j721e-auto-common.dtso b/arch/arm64/boot/dts/ti/k3-j721e-auto-common.dtso
new file mode 100644
index 00000000..dc0c835
--- /dev/null
+++ b/arch/arm64/boot/dts/ti/k3-j721e-auto-common.dtso
@@ -0,0 +1,182 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Auto-Processors: common device-tree overlay
+ *
+ * Copyright (C) 2019 Texas Instruments Incorporated - http://www.ti.com/
+ */
+
+/dts-v1/;
+/plugin/;
+
+/ {
+	fragment@101 {
+		target = <&mcu_r5fss0_core0_dma_memory_region>;
+		#address-cells = <2>;
+		#size-cells = <2>;
+		__overlay__ {
+			reg = <0x00 0xa0000000 0x00 0x00100000>;
+		};
+	};
+
+	fragment@102 {
+		target = <&mcu_r5fss0_core0_memory_region>;
+		#address-cells = <2>;
+		#size-cells = <2>;
+		__overlay__ {
+			reg = <0x00 0xa0100000 0x00 0x00f00000>;
+		};
+	};
+
+	fragment@103 {
+		target = <&mcu_r5fss0_core1_dma_memory_region>;
+		#address-cells = <2>;
+		#size-cells = <2>;
+		__overlay__ {
+			reg = <0x00 0xa1000000 0x00 0x00100000>;
+		};
+	};
+
+	fragment@104 {
+		target = <&mcu_r5fss0_core1_memory_region>;
+		#address-cells = <2>;
+		#size-cells = <2>;
+		__overlay__ {
+			reg = <0x00 0xa1100000 0x00 0x00f00000>;
+		};
+	};
+
+	fragment@105 {
+		target = <&main_r5fss0_core0_dma_memory_region>;
+		#address-cells = <2>;
+		#size-cells = <2>;
+		__overlay__ {
+			reg = <0x00 0xa2000000 0x00 0x00100000>;
+		};
+	};
+
+	fragment@106 {
+		target = <&main_r5fss0_core0_memory_region>;
+		#address-cells = <2>;
+		#size-cells = <2>;
+		__overlay__ {
+			reg = <0x00 0xa2100000 0x00 0x01f00000>;
+		};
+	};
+
+	fragment@107 {
+		target = <&main_r5fss0_core1_dma_memory_region>;
+		#address-cells = <2>;
+		#size-cells = <2>;
+		__overlay__ {
+			reg = <0x00 0xa4000000 0x00 0x00100000>;
+		};
+	};
+
+	fragment@108 {
+		target = <&main_r5fss0_core1_memory_region>;
+		#address-cells = <2>;
+		#size-cells = <2>;
+		__overlay__ {
+			reg = <0x00 0xa4100000 0x00 0x03f00000>;
+		};
+	};
+
+	fragment@109 {
+		target = <&main_r5fss1_core0_dma_memory_region>;
+		#address-cells = <2>;
+		#size-cells = <2>;
+		__overlay__ {
+			reg = <0x00 0xa8000000 0x00 0x00100000>;
+		};
+	};
+
+	fragment@110 {
+		target = <&main_r5fss1_core0_memory_region>;
+		#address-cells = <2>;
+		#size-cells = <2>;
+		__overlay__ {
+			reg = <0x00 0xa8100000 0x00 0x00f00000>;
+		};
+	};
+
+	fragment@111 {
+		target = <&main_r5fss1_core1_dma_memory_region>;
+		#address-cells = <2>;
+		#size-cells = <2>;
+		__overlay__ {
+			reg = <0x00 0xa9000000 0x00 0x00100000>;
+		};
+	};
+
+	fragment@112 {
+		target = <&main_r5fss1_core1_memory_region>;
+		#address-cells = <2>;
+		#size-cells = <2>;
+		__overlay__ {
+			reg = <0x00 0xa9100000 0x00 0x00f00000>;
+		};
+	};
+
+	fragment@113 {
+		target = <&c66_1_dma_memory_region>;
+		#address-cells = <2>;
+		#size-cells = <2>;
+		__overlay__ {
+			reg = <0x00 0xaa000000 0x00 0x00100000>;
+		};
+	};
+
+	fragment@114 {
+		target = <&c66_0_memory_region>;
+		#address-cells = <2>;
+		#size-cells = <2>;
+		__overlay__ {
+			reg = <0x00 0xaa100000 0x00 0x03f00000>;
+		};
+	};
+
+	fragment@115 {
+		target = <&c66_0_dma_memory_region>;
+		#address-cells = <2>;
+		#size-cells = <2>;
+		__overlay__ {
+			reg = <0x00 0xae000000 0x00 0x00100000>;
+		};
+	};
+
+	fragment@116 {
+		target = <&c66_1_memory_region>;
+		#address-cells = <2>;
+		#size-cells = <2>;
+		__overlay__ {
+			reg = <0x00 0xae100000 0x00 0x01f00000>;
+		};
+	};
+
+	fragment@117 {
+		target = <&c71_0_dma_memory_region>;
+		#address-cells = <2>;
+		#size-cells = <2>;
+		__overlay__ {
+			reg = <0x00 0xb0000000 0x00 0x00100000>;
+		};
+	};
+
+	fragment@118 {
+		target = <&c71_0_memory_region>;
+		#address-cells = <2>;
+		#size-cells = <2>;
+		__overlay__ {
+			reg = <0x00 0xb0100000 0x00 0x07f00000>;
+		};
+	};
+
+	fragment@119 {
+		target = <&rtos_ipc_memory_region>;
+		#address-cells = <2>;
+		#size-cells = <2>;
+		__overlay__ {
+			reg = <0x00 0xba000000 0x00 0x02000000>;
+		};
+	};
+};
diff --git a/arch/arm64/boot/dts/ti/k3-j721e-psdkla-apps.dtso b/arch/arm64/boot/dts/ti/k3-j721e-psdkla-apps.dtso
new file mode 100644
index 00000000..2779582
--- /dev/null
+++ b/arch/arm64/boot/dts/ti/k3-j721e-psdkla-apps.dtso
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: GPL-2.0
+/**
+ * PSDKLA specific DT customizations
+ *
+ * Copyright (C) 2019 Texas Instruments Incorporated - http://www.ti.com/
+ */
+
+/dts-v1/;
+/plugin/;
+
+&main_uart2 {
+	status = "disabled";
+};
+
diff --git a/arch/arm64/boot/dts/ti/k3-j721e-vision-apps.dtso b/arch/arm64/boot/dts/ti/k3-j721e-vision-apps.dtso
new file mode 100644
index 00000000..9c899d1
--- /dev/null
+++ b/arch/arm64/boot/dts/ti/k3-j721e-vision-apps.dtso
@@ -0,0 +1,190 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Vision-apps: device-tree overlay
+ *
+ * Copyright (C) 2019 Texas Instruments Incorporated - http://www.ti.com/
+ */
+
+#include <dt-bindings/soc/ti,sci_pm_domain.h>
+
+/dts-v1/;
+/plugin/;
+
+/ {
+	fragment@101 {
+		target-path = "/";
+
+		__overlay__ {
+
+			dummy-panel {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				compatible = "ti,vx-app-dummy";
+
+				port@0 {
+					reg = <0>;
+
+					dummy_in: endpoint {
+						remote-endpoint = <&dpi_out3>;
+					};
+				};
+
+			};
+
+			cmem {
+				compatible = "ti,cmem";
+				#address-cells = <1>;
+				#size-cells = <0>;
+				status = "okay";
+				cmem_block_0: cmem_block@0 {
+					reg = <0>;
+					memory-region = <&cmem_shared_region>;
+				};
+			};
+		};
+	};
+
+	fragment@102 {
+		target-path = "/chosen";
+
+		__overlay__ {
+			gl-use-nulldrmws;
+		};
+	};
+};
+
+&reserved_memory {
+	#address-cells = <2>;
+	#size-cells = <2>;
+
+	vision_apps_memory_region: vision_apps-dma-memory@b8000000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xb8000000 0x00 0x02000000>;
+		no-map;
+	};
+
+	cmem_shared_region: cmem-memories@bc000000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xbc000000 0x00 0x24000000>;
+		no-map;
+	};
+};
+
+&main_i2c6 {
+	status = "disabled";
+};
+
+&serdes_wiz4 {
+	status = "disabled";
+};
+
+&mhdp {
+	status = "disabled";
+};
+
+&dss {
+	status = "ok";
+
+	pinctrl-names = "none"; /* pinmux configured by RTOS */
+	power-domains = <&k3_pds 152 TI_SCI_PD_SHARED>; /* share IP among VMs and RTOS */
+
+	/* No changes to parents or rates for VP clocks
+	 * if the VP is not owned by us
+	 */
+	assigned-clocks = <&k3_clks 152 13>;
+
+	assigned-clock-parents = <&k3_clks 152 18>;	/* PLL23_HSDIV0 */
+
+	/* partition information */
+	dss_planes: dss-planes {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		/* vid1, marshalled to us by RTOS */
+		plane@0 {
+			reg = <0>;
+			managed = <0>;
+		};
+
+		/* vidl1, Reserved for jailhouse inmate */
+		plane@1 {
+			reg = <1>;
+			managed = <0>;
+		};
+
+		/* vid2, owned by RTOS */
+		plane@2 {
+			reg = <2>;
+			managed = <0>;
+		};
+
+		/* vidl2, marshalled to us by RTOS */
+		plane@3 {
+			reg = <3>;
+			managed = <1>;
+		};
+	};
+
+	dss_vps: dss-vps {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		/* Owned by jailhouse inmate */
+		vp@0 {
+			reg = <0>;
+			managed = <0>;
+		};
+
+		/* Owned by RTOS */
+		vp@1 {
+			reg = <1>;
+			managed = <0>;
+		};
+
+		/* The 2 below are not owned by anyone
+		 * else, so keeping here
+		 */
+		vp@2 {
+			reg = <2>;
+			managed = <0>;
+		};
+
+		vp@3 {
+			reg = <3>;
+			managed = <1>;
+		};
+	};
+
+	dss_commons: dss-commons {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		interrupt-common {
+			reg = <1>;
+		};
+
+		config-common {
+			status = "disabled";
+			reg = <0>;
+		};
+	};
+};
+
+&dss_ports {
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	port@3 {
+		reg = <3>;
+
+		dpi_out3: endpoint {
+			remote-endpoint = <&dummy_in>;
+		};
+	};
+};
+
+&main_i2c1 {
+	status = "disabled";
+};
+
+
-- 
1.9.1

